packages/scene-composer/src/components/panels/scene-rule-components/SceneRuleTargetEditor.tsx

```diff
@@ -92,23 +102,47 @@ export const SceneRuleTargetEditor: React.FC<ISceneRuleTargetEditorProps> = ({
           <SceneRuleTargetIconEditor
             targetValue={targetInfo.value}
             onChange={(targetValue) => {
-              const colorWithIcon = targetValue === 'Custom' ? `${targetValue}-${chosenColor}` : targetValue;
-              onChange(convertToIotTwinMakerNamespace(targetInfo.type, colorWithIcon));
+              onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetValue));
             }}
             chosenColor={chosenColor}
+            customIcon={icon as IconLookup}
           />
           {isCustomStyle && (
-            <ColorPicker
-              color={chosenColor}
-              onSelectColor={(newColor) => {
-                const colorWithIcon =
-                  targetInfo.value === 'Custom' ? `${targetInfo.value}-${newColor}` : targetInfo.value;
-                onChange(convertToIotTwinMakerNamespace(targetInfo.type, colorWithIcon));
-                setChosenColor(newColor);
-              }}
-              colorPickerLabel={formatMessage({ defaultMessage: 'Colors', description: 'Colors' })}
-              customColorLabel={formatMessage({ defaultMessage: 'Custom colors', description: 'Custom colors' })}
-            />
+            <>
+              <ColorPicker
+                color={chosenColor}
+                onSelectColor={(newColor) => {
+                  onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetInfo.value), {
+                    ...targetMetadata,
+                    color: newColor,
+                  } as TargetMetadata<string>);
+                  setChosenColor(newColor);
+                }}
+                customColors={tagStyleColors}
+                onUpdateCustomColors={(customColors) =>
+                  setSceneProperty(KnownSceneProperty.TagCustomColors, customColors)
+                }
+                colorPickerLabel={formatMessage({ defaultMessage: 'Colors', description: 'Colors' })}
+                customColorLabel={formatMessage({ defaultMessage: 'Custom colors', description: 'Custom colors' })}
+              />
+              <IconPicker
+                onSelectIconChange={(pickedIcon) => {
+                  onChange(convertToIotTwinMakerNamespace(targetInfo.type, targetInfo.value), {
+                    ...targetMetadata,
+                    iconPrefix: pickedIcon.prefix,
+                    iconName: pickedIcon.iconName,
+                  } as TargetMetadata<string>);
+                  setIcon(pickedIcon as IconLookup);
+                }}
+                selectedIcon={icon as IconLookup}
```

Comment:
do you need to cast to `IconLookup` here? can we use `IIconLookup` for any internal component, then only cast to `IconLookup` when passing it to fontawsome?

Reply:
No we don't